### ISSUE TITLE
XHR request using CORS always returns a 403 Forbidden when response content-type is application/json

### ISSUE THREAD

**Issue by brandonweiss** [cyu/rack-cors#7]
To start, I don't think this is actually a bug in `rack-cors`, but the problem I'm having seems to be rather esoteric and possibly related to CORS, so this seemed like a good place to start.

The scenario is, I'm building an app with a Backbone.js frontend and a Sinatra api/backend. Initially they were all mounted inside the same rack app, but I wanted to break them out into separate apps, with the api at `api.someurl.com` and the frontend at `someurl.com`. So I started brushing up on CORS. I attempted to just set the appropriate headers myself in a before/after filter in Sinatra, but no matter what I did Chrome wouldn't allow the request. Eventually I found `rack-cors`, implemented it like so:

```
use Rack::Cors do
  allow do
    origins  "*"
    resource "*", headers: :any, methods: [:get, :post, :options]
  end
end
```

And suddenly Chrome started making the cross-domain request. I looked at the CORS headers that `rack-cors` generates and they were exactly the same ones I'd been setting manually, and yet, inexplicably they didn't work (when I set them manually). Out of curiosity, if you happen to have any idea why that might be, I'd love to know.

But, even though the request was now going through, it was actually failing with a `403 Forbidden`. The request I was making was a really basic request; I just wanted to see the request/response headers and body in Chrome inspector. This was it:

```
<script>
  var xhr = new XMLHttpRequest();
  xhr.open("GET", "http://api.someurl.com", true);
  xhr.setRequestHeader("Content-Type", "application/json");
  xhr.send();
</script>
```

And it was generating a `403 Forbidden`. Interestingly, the response content-type was `text/plain`. Which is weird, because it's a pure JSON api, so I globally override the content-type on all responses to `application/json` in a before/after filter, like so:

```
before do
  response["Content-Type"] = "application/json"
end
```

I was pretty stumped, so I started changing the request/response content-types to see what would happen. If I remove the content-type override (which the response then defaults to `text/html`) or set it to any other content-type than `application/json`, it works. I get a 200 and the correct data back. The actual request content-type seems to be irrelevant. I changed the request to `text/html`, set the response to `application/json`, and I get a `403 Forbidden`. But, if I view that same resource in the browser, which requests with `text/html`, then I get a 200 response, the JSON data, and the content-type is `application/json`.

So basically, there's something about making a CORS XHR request to a Sinatra/rack app using `rack-cors`, with the content-type for responses set to `application/json`, that causes the response to hork and return a 403 Forbidden.

Any help would be much appreciated. Thanks!
